dipper/sources/UCSCBands.py

```diff
@@ -151,73 +211,76 @@ def __init__(self,
-        if self.tax_ids is None:
-            # self.tax_ids = [9606, 10090, 7955]
+        if self.tax_ids is None or self.tax_ids == '':
             self.species = (
-                'Homo sapiens',       # 9606,
-                'Mus musculus',       # 10090,
-                'Danio rerio',        # 7955,
-                'Bos taurus',         # 9913,
-                'Gallus gallus',      # 9031,
-                'Sus scrofa',         # 9823,
-                'Ovis aries',         # 9940,
-                'Equus caballus'      # 9796
+                'Homo sapiens',             # NCBITaxon:9606,
+                'Mus musculus',             # NCBITaxon:10090,
+                'Danio rerio',              # NCBITaxon:7955,
+                'Bos taurus',               # NCBITaxon:9913,
+                'Gallus gallus',            # NCBITaxon:9031,
+                'Sus scrofa',               # NCBITaxon:9823,
+                'Ovis aries',               # NCBITaxon:9940,
+                'Equus caballus',           # NCBITaxon:9796,
+                'Canis lupus familiaris',   # NCBITaxon:9615,
+                'Felis catus',              # NCBITaxon:9685
+                # TODO add other species as defaults  ... still?
             )
-
-            self.tax_ids = (self.localtt[x] for x in self.species)
+            self.tax_ids = [self.localtt[x] for x in self.species]
         else:
-            self.species = (self.localtcid[x] for x in self.tax_ids)
-
-        # TODO add other species as defaults
+            self.species = [self.localtcid[x] for x in self.tax_ids]
 
+        # note:
+        # must map UCSC tags to their NCBI equivalent in the local translation table
         self._check_tax_ids()
 
         # data-source specific warnings
         # (will be removed when issues are cleared)
 
     def fetch(self, is_dl_forced=False):
-
         self.get_files(is_dl_forced)
 
     def parse(self, limit=None):
 
         if limit is not None:
             LOG.info("Only parsing first %d rows", limit)
 
-        LOG.info("Parsing files...")
+        LOG.info("Parsing files for %s", str(self.tax_ids))
 
         if self.test_only:
             self.test_mode = True
 
         for taxon in self.tax_ids:
-            self._get_chrbands(limit, taxon, 'UCSC:'+self.files[taxon]['build_num'])
+            LOG.info('\tParsing  cytoBandIdeo file for %s', taxon)
+            self._get_chrbands(limit, taxon, 'UCSC:' + self.files[taxon]['build_num'])
 
+        LOG.info('Creating genome build statments')
         self._create_genome_builds()
 
         # using the full graph as the test here
         self.testgraph = self.graph
         LOG.info("Done parsing files.")
 
-    def _get_chrbands(self, limit, taxon, genome_id):
+    def _get_chrbands(self, limit, src_key, genome_id):
         """
         :param limit:
         :return:
 
         """
-
+        tax_num = src_key
+        species_name = self.globaltcid[tax_num]  # for logging
```

Comment:
Getting a key error here, this I think should be `self.globaltcid["NCBITaxon:" + tax_num]`

Reply:
thanks @justaddcoffee  good catch